eth/blockwatch/block_watcher.go

```diff
@@ -159,7 +159,12 @@ func (w *Watcher) syncToLatestBlock() error {
 		return err
 	}
 
-	for i := lastSeenHeader.Number; i.Cmp(newestHeader.Number) < 0; i = i.Add(i, big.NewInt(1)) {
+	lastSeenHeaderNumber := big.NewInt(w.startBlockDepth.Int64())
```

Comment:
On second thought given the current configuration of the block watcher I don't think iterating starting from startBlockDepth leads to the correct desired behavior.

First, some acknowledgements. Typically we would not expect to set lastSeenHeaderNumber to startBlockDepth because syncToLatestBlock() is called in the watch loop started by Watch() which is always called after backfilling and backfilling would ensure that there is a block stored in the DB. 

*If* we happen to get to this case for some reason, then the current code would set lastSeenHeaderNumber to  startBlockDepth which is set to [rpc.LatestBlockNumber](https://github.com/livepeer/go-livepeer/blob/4e24711bfabc873f1be7e98cdd8599f68a6cfedd/cmd/livepeer/livepeer.go#L436). Since `rpc.LatestBlockNumber` is an [alias for -1](https://github.com/ethereum/go-ethereum/blob/053ed9cc847647a9b3ef707d0efe7104c4ab2a4c/rpc/types.go#L73) the loop on L167 would start at -1 and end at newestHeader.Number which is not the behavior we want.

A startBlockDepth of `rpc.LatestBlockNumber` means that we just want to fetch the latest block since passing `rpc.LatestBlockNumber` to the `eth_*` RPC calls will return data on the latest block. So, I think the correct behavior here would be something like:

```golang
if lastSeenHeader == nil {
    return w.pollNextBlock() 
}

for i := lastSeenHeader.Number; i.Cmp(newestHeader.Number) < 0; i = i.Add(i, big.NewInt(1)) { ... }
```

`pollNextBlock()` should know to fetch the latest block [when there are no blocks in the DB](https://github.com/livepeer/go-livepeer/blob/b909883c2535a434dfc4cd5078d8c249df9a1f18/eth/blockwatch/block_watcher.go#L181).